File: swift/Sources/AspectlyBridge/BridgeLogger.swift
import Foundation

/// Simple logging interface for `BridgeHost`.
/// Implement this to integrate with your logging framework.
public protocol BridgeLogger {
    /// Log a debug message.
    func debug(_ message: String)
    /// Log an info message.
    func info(_ message: String)
    /// Log a warning message.
    func warn(_ message: String)
    /// Log an error message with an optional underlying error.
    func error(_ message: String, _ error: Error?)
}

public extension BridgeLogger {
    func error(_ message: String) {
        error(message, nil)
    }
}

/// Default logger implementation that writes to the console.
public struct ConsoleLogger: BridgeLogger {
    public static let shared = ConsoleLogger()

    public init() {}

    public func debug(_ message: String) {
        print("[DEBUG] \(message)")
    }

    public func info(_ message: String) {
        print("[INFO] \(message)")
    }

    public func warn(_ message: String) {
        print("[WARN] \(message)")
    }

    public func error(_ message: String, _ error: Error?) {
        print("[ERROR] \(message)")
        if let error = error {
            print("  Error: \(error)")
        }
    }
}

/// Null logger that discards all messages (no-op).
public struct NullLogger: BridgeLogger {
    public static let shared = NullLogger()

    public init() {}

    public func debug(_ message: String) {}
    public func info(_ message: String) {}
    public func warn(_ message: String) {}
    public func error(_ message: String, _ error: Error?) {}
}

File: android/aspectly-bridge/src/main/kotlin/com/aspectly/bridge/BridgeLogger.kt
package com.aspectly.bridge

/**
 * Simple logging interface for [BridgeHost].
 * Implement this to integrate with your logging framework.
 */
interface BridgeLogger {
    fun debug(message: String)
    fun info(message: String)
    fun warn(message: String)
    fun error(message: String, error: Throwable? = null)
}

/** Default logger implementation that writes to standard output. */
object ConsoleLogger : BridgeLogger {
    override fun debug(message: String) = println("[DEBUG] $message")
    override fun info(message: String) = println("[INFO] $message")
    override fun warn(message: String) = println("[WARN] $message")
    override fun error(message: String, error: Throwable?) {
        println("[ERROR] $message")
        if (error != null) println("  Error: $error")
    }
}

/** Null logger that discards all messages (no-op). */
object NullLogger : BridgeLogger {
    override fun debug(message: String) {}
    override fun info(message: String) {}
    override fun warn(message: String) {}
    override fun error(message: String, error: Throwable?) {}
}
